src/json.rs

```diff
@@ -253,10 +290,18 @@ pub struct JsonBody<T: HttpMessage, U: DeserializeOwned> {
 
 impl<T: HttpMessage, U: DeserializeOwned> JsonBody<T, U> {
     /// Create `JsonBody` for request.
-    pub fn new(req: &T) -> Self {
+    pub fn new<S>(req: &T, cfg: Option<&JsonConfig<S>>) -> Self {
         // check content-type
         let json = if let Ok(Some(mime)) = req.mime_type() {
-            mime.subtype() == mime::JSON || mime.suffix() == Some(mime::JSON)
+            if let Some(cfg) = cfg {
+                cfg.content_types.iter().any(|ct| {
+                    (ct.type_.is_none() || ct.type_.unwrap() == mime.type_()) &&
+                        (ct.subtype.is_none() || ct.subtype.unwrap() == mime.subtype()) &&
+                        (ct.suffix.is_none() || ct.suffix == mime.suffix())
+                })
+            } else {
+                mime.subtype() == mime::JSON || mime.suffix() == Some(mime::JSON)
```

Comment:
This condition should be checked first before going into `if let Some(cfg) = cfg`

I think it would be better to treat extra content types as extra.